client/src/main/java/tds/student/Student.java

```diff
@@ -4,37 +4,36 @@
  * Represents a student taking an exam
  */
 public class Student {
-    private long id;
-    private String studentId;
-    private String stateCode;
-    private String clientName;
+    private final long id;
+    private final String externalStudentId;
```

Comment:
I actually kinda think `externalStudentId` is the most appropriate name, since it could be the SSID or Alternate SSID (depending on how the system is configured).  My first instinct woulda been to call it `studentId`, but my mind (wrongly) translates that directly to SSID.  Since this field can have one of two meanings depending on system configuration, the best alternative to `externalStudentId` I can come up with is `studentKey`.